include/fx/gltf.h

```diff
@@ -1839,46 +1878,156 @@ namespace gltf
+                    if (!is.good())
+                    {
+                        throw std::system_error(std::make_error_code(std::errc::io_error));
+                    }
+                }
+                while (!is.eof())
+                {
+                    detail::ChunkHeader head;
+                    is.read(reinterpret_cast<char *>(&head), detail::ChunkHeaderSize);
+                    if (!is.eof())
+                    {
+                        if (!is.good())
+                        {
+                            throw std::system_error(std::make_error_code(std::errc::io_error));
+                        }
+                        totalSize += detail::ChunkHeaderSize + head.chunkLength;
+                        if (totalSize > readQuotas.MaxFileSize)
+                        {
+                            throw invalid_gltf_document("Quota exceeded : file size > MaxFileSize");
+                        }
+                        extraChunks.resize(extraChunksLength + detail::ChunkHeaderSize + head.chunkLength);
+                        memcpy(&extraChunks[extraChunksLength], &head, detail::ChunkHeaderSize);
+                        is.read(reinterpret_cast<char *>(&extraChunks[extraChunksLength + detail::ChunkHeaderSize]), head.chunkLength);
+                        if (!is.good())
+                        {
+                            throw std::system_error(std::make_error_code(std::errc::io_error));
+                        }
+                        extraChunksLength += detail::ChunkHeaderSize + head.chunkLength;
+                    }
+                }
             }
 
-            detail::GLBHeader header;
-            std::memcpy(&header, &binary[0], detail::HeaderSize);
-            if (header.magic != detail::GLBHeaderMagic ||
-                header.jsonHeader.chunkType != detail::GLBChunkJSON ||
-                header.jsonHeader.chunkLength + detail::HeaderSize > header.length ||
-                header.jsonHeader.chunkLength + detail::HeaderSize > binary.size())
+            return detail::Create(
+                nlohmann::json::parse({ &binary[detail::HeaderSize], header.jsonHeader.chunkLength }),
+                { bufferRootPath, readQuotas, &binary, detail::HeaderSize + header.jsonHeader.chunkLength,
+                  &extraChunks, extraChunksLength });
+        }
+        catch (invalid_gltf_document &)
+        {
+            throw;
+        }
+        catch (std::system_error &)
+        {
+            throw;
+        }
+        catch (...)
+        {
+            std::throw_with_nested(invalid_gltf_document("Invalid glTF document. See nested exception for details."));
+        }
+    }
+
+    inline Document LoadFromBinary(std::string const & documentFilePath, ReadQuotas const & readQuotas = {})
+    {
+        try
+        {
+            std::ifstream file(documentFilePath, std::ios::binary);
+            if (!file.is_open())

```

Comment:
This check should be moved into LoadFromBinaryStream.  Then the try/catches here can all be removed and let LoadFromBinaryStream handle everything.